### ISSUE TITLE
Provide setting to organize imports on format

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#48547]
Testing #48150

This would be awesome.

**Comment by daBishMan** [comment]
I know we have the setting for organize import on saving, on import will be great, but can we please have it respect the formatting and not create a line over 140 chars wide, or a setting to define that please?

**Comment by quicksnap** [comment]
Would love this action to happen _before_ formatting, since may users use `prettier`. Currently, I organize imports (which does not respect prettier formatting rules) and then format with prettier (manually or onSave)

**Comment by ZSkycat** [comment]
The Typescript Hero has a function to set the import group, but it is still relatively simple and can be used as a reference.
https://github.com/buehler/typescript-hero/wiki/Features
https://github.com/buehler/typescript-hero/issues/462

**Comment by Springrbua** [comment]
I agree with @ZSkycat. I am currently using TypeScript Hero, but since most of the features are now included in TypeScript itself, I would like to work without that extension.
The only thing that seems to be missing right now are some features of the "organize imports".
For instance, I really like the grouped imports, which in my case create a group for my modules (relative paths) and a group for libraries (absolute path).

**Comment by Xaz16** [comment]
Hi guys

It is really annoying that my prettier have conflict with auto organize imports.
@mjbvz 
Could you please tell if you have any progress on it?

As workaround I use 
```json
"editor.formatOnSaveTimeout": 1000
```
to give prettier a chance to make correct formatting.
